modules/core/include/opencv2/core/mat.hpp

```diff
@@ -358,13 +361,15 @@ class CV_EXPORTS _InputOutputArray : public _OutputArray
     template<typename _Tp, int m, int n> _InputOutputArray(const Matx<_Tp, m, n>& matx);
     _InputOutputArray(const UMat& m);
     _InputOutputArray(const std::vector<UMat>& vec);
+
+    explicit _InputOutputArray(const _OutputArray& o);
 };
 
-typedef const _InputArray& InputArray;
+typedef _InputArray InputArray;
```

Comment:
Are you sure?

I see these advantages/disadvantages of appraches:
- old style (reference): no copy call on parameter propagation to nested functions (+), additional pointer dereference for field method access in inlined methods (-)
- new style (instance): object copy on parameter propagation to nested functions (-), direct access to field values in inlined methods (+)